index.js

```diff
@@ -10,41 +10,79 @@ const path = require("path");
 const loaderUtils = require("loader-utils");
 const SourceNode = require("source-map").SourceNode;
 const SourceMapConsumer = require("source-map").SourceMapConsumer;
-const FOOTER = "/*** REQUIRES FROM autoload-plugins-loader ***/\n";
+const matchAll = require('match-all');
 
-module.exports = function(content, sourceMap) {
-    const resourcePath = this.resourcePath;
+const FOOTER = "/*** REQUIRES FROM autoload-modules-loader ***/\n";
+
+const REQUIRE_RE = /\brequire\s*\(\s*[\'"]([^\'"]*)[\'"]\s*\)/g;
 
+module.exports = function(content, sourceMap) {
     if (this.cacheable) {
         this.cacheable();
     }
 
-    const query = loaderUtils.getOptions(this) || {};
-    const requires = [];
-    const keys = Object.keys(query);
-
-    if (keys.length == 1 && typeof query[keys[0]] == "boolean") {
-        const mod = keys[0];
-        const modPath = path.resolve(__dirname, mod);
-        requires.push(`require("${modPath}");   // ${mod}`);
-    } else {
-        keys.forEach(function(name) {
-            const mod = name;
-            const modPath = path.resolve(__dirname, mod);
-            requires.push(`require("${modPath}");   // ${mod}`);
+    // promisify-node doesn't know how to deal with methods so we have to wrap
+    // this.resolve() ourselves.
+    const resolve = (context, request) =>
+        new Promise((resolve, reject) => {
+            this.resolve(context, request, (err, result) => {
+                if (err) {
+                    reject(err);
+                } else {
+                    resolve(result);
+                }
+            });
         });
-    }
 
-    if (sourceMap) {
-        const currentRequest = loaderUtils.getCurrentRequest(this);
-        const node = SourceNode.fromStringWithSourceMap(content, new SourceMapConsumer(sourceMap));
-        node.add("\n\n" + FOOTER + requires.join("\n"));
-        const result = node.toStringWithSourceMap({
-            file: currentRequest
-        });
-        this.callback(null, result.code, result.map.toJSON());
-        return;
-    }
+    // Find all 'require' statements
+    const reqs = matchAll(content, REQUIRE_RE).toArray();
```

Comment:
To clarify: this is all happening post-babel, right?

Reply:
Yes.  I'll make sure there are comments in webapp's webpack.config.js.  It's easy to control the order in which loaders are run.

Reply:
I'll also add a note to the README that this should be run after transpiling ES6 b/c it can't handle `import` statements.  I'll open an issue for this, but it's not necessary for this to handle our use case.